tools/topology/topology1/CMakeLists.txt

```diff
@@ -329,11 +335,16 @@ foreach(tplg ${TPLGS})
 			${CMAKE_CURRENT_SOURCE_DIR}/${input}.m4
 			> production/${output}.conf
 		MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/${input}.m4
-		DEPENDS abi_v1
+		# The 'abi.h' file is the real dependency that triggers
+		# a rebuild. 'abi_v1' is the target that avoids a race
+		# to rebuild abi.h. See the CMake FAQ or (better) Sam
+		# Thursfield's blog about custom targets.
+		DEPENDS abi_v1 ${CMAKE_CURRENT_BINARY_DIR}/abi.h
```

Comment:
Have you tried using [set_source_files_properties](https://cmake.org/cmake/help/latest/command/set_source_files_properties.html) and setting [GENERATED](https://cmake.org/cmake/help/latest/prop_sf/GENERATED.html) on abi.h?
I did not analyzed whole dependency system here, maybe it can't be used -  just a suggestion.

Reply:
Thanks for the review! No I did not trys that.

https://cmake.org/cmake/help/latest/prop_sf/GENERATED.html says GENERATED is automatically set but... works only in the current directory.

I followed https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/ and tested it quite thoroughly. Seems OK.